### PR TITLE
Add support for shutdown and startup functions

### PR DESCRIPTION
**Pull request by TheBurchLog**

### CODE REVIEW

**Comment by TheBurchLog** [comment]
Example Plugin

```yaml
PLUGIN_ENTRY="-m shutdown"
CLIENT_SHUTDOWN_FUNCTION="shutdown_two"
CLIENT_STARTUP_FUNCTION="startup_two"
```

```python
from brewtils import Plugin, command, shutdown, startup

__version__ = "3.0.0.dev0"

class ShutdownClient:

    @command
    def my_function(self):
        return "my_function"
    
    @shutdown
    def shutdown_one(self):
        print("Internal Shutdown One Function")

    def shutdown_two(self):
        print("Internal Shutdown Two Function")

    @startup
    def startup_one(self):
        print("Internal Startup One Function")

    def startup_two(self):
        print("Internal Startup Two Function")

def main():
    def other_shutdown_function():
        print("External Shutdown Function")

    def other_startup_function():
        print("External Startup Function")

    Plugin(
        ShutdownClient(),
        name="shutdown_example",
        version=__version__,
        shutdown_function=other_shutdown_function,
        startup_function=other_startup_function,
    ).run()


if __name__ == "__main__":
    main()
```

File: src/app/beer_garden/local_plugins/manager.py [hunk 1]
```diff
@@ -706,6 +706,38 @@ def _process_args(plugin_config: Dict[str, Any], instance_name: str):
             if plugin_group is not None:
                 process_args += ["--group=" + str(plugin_group)]
 
+        if plugin_config.get("CLIENT_SHUTDOWN_FUNCTIONS") or plugin_config.get(
+            "CLIENT_SHUTDOWN_FUNCTIONS"
```
Comment (1maple1) [inline src/app/beer_garden/local_plugins/manager.py:710]: This is the same condition

File: src/app/beer_garden/local_plugins/manager.py [hunk 2]
```diff
@@ -706,6 +706,38 @@ def _process_args(plugin_config: Dict[str, Any], instance_name: str):
...
+                shutdown_functions.append(
+                    plugin_config.get("CLIENT_SHUTDOWN_FUNCTIONS")
+                )
+
+            process_args += [
+                "--client_shutdown_functions=" + arg for arg in shutdown_functions
+            ]
+
+        if plugin_config.get("CLIENT_STARTUP_FUNCTIONS") or plugin_config.get(
+            "CLIENT_STARTUP_FUNCTIONS"
```
Comment (1maple1) [inline src/app/beer_garden/local_plugins/manager.py]: This is the same condition

**Review by 1maple1 — APPROVED** [review]
Just the two comments, otherwise looks good.

**Pull request merged**